addons/mail/models/mail_channel.py

```diff
@@ -232,6 +232,15 @@ def message_post(self, body='', subject=None, message_type='notification', subty
         message = super(Channel, self.with_context(mail_create_nosubscribe=True)).message_post(body=body, subject=subject, message_type=message_type, subtype=subtype, parent_id=parent_id, attachments=attachments, content_subtype=content_subtype, **kwargs)
         return message
 
+    def _alias_check_contact(self, message, message_dict, alias):
+        if alias.alias_contact == 'followers' and self.ids:
+            author = self.env['res.partner'].browse(message_dict.get('author_id', False))
+            if not author or author not in self.channel_partner_ids:
+                return {
+                    'error_mesage': _('restricted to channel members'),
+                }
+        return super(Channel, self)._alias_check_contact(message, message_dict, alias)
```

Comment:
IMHO: super before, check followers, or return result of super. Otherwise followers key not correctly computed.